plugin/ContextMenuHook.qml

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2015 Caliste Damien.
+ * Contact: Damien Caliste <dcaliste@free.fr>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; version 2 only.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import QtQuick 2.0
+import Sailfish.Silica 1.0
+
+Item {
+    id: hook
+
+    property alias backgroundColor: background.color
+    property alias backgroundOpacity: background.opacity
+
+    property variant _menu
+    property real _height0
+
+    // Used to emulate the MouseArea that trigger a ContextMenu
+    property bool pressed: true
+    property bool preventStealing
+
+    signal positionChanged()
+    signal released()
+
+    function showMenuForRect(menu, rect) {
+        y = rect.y
+        _height0 = rect.height + Theme.paddingSmall
+        _menu = menu
+        menu.show(contextHook)
+    }
+
+    width: parent.width
+    height: _height0 + (_menu ? _menu.height : 0.)
+
+    Rectangle {
+        id: background
+        parent: _menu ? _menu : null
+        anchors.fill: parent ? parent : undefined
+        color: Theme.highlightDimmerColor
+        opacity: 0.91
+        Binding on z {
+            when: parent
+            value: parent.z - 1
```

Comment:
I think z: -1 should provide the same result as this binding. Negative values stacked behind parent.